lib/galaxy/schema/help.py

```diff
@@ -0,0 +1,107 @@
+from typing import (
+    Any,
+    List,
+    Optional,
+)
+
+from pydantic import (
+    Extra,
+    Field,
+)
+from typing_extensions import Annotated
+
+from galaxy.schema.schema import Model
+
+
+# TODO: remove this class once we have a proper model for all temp models
+class HelpTempBaseModel(Model):
+    class Config:
+        extra = Extra.allow
+
+
+class HelpForumPost(HelpTempBaseModel):
+    """Model for a post in the help forum."""
+
+    id: Annotated[int, Field(description="The ID of the post.")]
+    name: Annotated[str, Field(description="The name of the post.")]
+    username: Annotated[str, Field(description="The username of the post author.")]
+    avatar_template: Annotated[str, Field(description="The avatar template of the user.")]
+    created_at: Annotated[str, Field(description="The creation date of the post.")]
+    like_count: Annotated[int, Field(description="The number of likes of the post.")]
+    blurb: Annotated[str, Field(description="The blurb of the post.")]
+    post_number: Annotated[int, Field(description="The post number of the post.")]
+    topic_id: Annotated[int, Field(description="The ID of the topic of the post.")]
+
+
+class HelpForumTopic(Model):
```

Comment:
Are these autogenerated ?

Reply:
I think they are not. Should they be?

Reply:
That would probably be wise, otherwise we'll have to manually figure out changes as the API evolves ? There's an OpenAPI schema in https://docs.discourse.org/openapi.json and we've used datamodel-codegen for instance to generate this: https://github.com/galaxyproject/galaxy/blob/520fa4a233145d18e3b2a53f49549ea9e4a65cc0/lib/galaxy/schema/drs/AccessMethod.py#L1

Reply:
Unfortunately, the schema provided by discourse is barely typed.
This is what we would be getting from the auto-generation:

```py
from typing import (
    List,
    Optional,
)

from pydantic import (
    BaseModel,
    Extra,
)


class GroupedSearchResult(BaseModel):
    class Config:
        extra = Extra.forbid

    more_posts: Optional[str]
    more_users: Optional[str]
    more_categories: Optional[str]
    term: str
    search_log_id: int
    more_full_page_results: Optional[str]
    can_create_topic: bool
    error: Optional[str]
    post_ids: List
    user_ids: List
    category_ids: List
    tag_ids: List
    group_ids: List


class JsonGetResponse(BaseModel):
    class Config:
        extra = Extra.forbid

    posts: List
    users: List
    categories: List
    tags: List
    groups: List
    grouped_search_result: GroupedSearchResult
```

Even though the types for posts, users, categories, etc. exist, they are not linked to the search response in the schema:

![image](https://github.com/galaxyproject/galaxy/assets/44241786/8f30859d-1da5-4e59-b2f1-76811dc64fc5)

We could use a mix, and extend the generated class with the correct types, or stick to the manual schema. Unsure what is best here.

Reply:
I suppose this depends on where the schema actually came from ... it's probably not the openapi document for lack of defined components? We're not enforcing anything on the backend, so i'm less worried about the correctness of the schema, but I wonder if this should even live together with the other pydantic models.

Reply:
For now if we can document how we got this and how one would go about updating it that'd be fine.